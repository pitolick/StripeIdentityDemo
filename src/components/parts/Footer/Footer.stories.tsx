import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { Footer } from './Footer'

const meta: Meta<typeof Footer> = {
  title: 'parts/Footer',
  component: Footer,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
}

export default meta
type Story = StoryObj<typeof Footer>

/**
 * デフォルト
 */
export const Default: Story = {}

/**
 * 子要素あり
 */
export const WithChildren: Story = {
  args: {
    children: <div className="container text-center">children</div>,
  },
}

/**
 * コピーライトのPropsあり
 */
export const WithCopyRightProps: Story = {
  args: {
    children: <div className="container text-center">children</div>,
  },
}
