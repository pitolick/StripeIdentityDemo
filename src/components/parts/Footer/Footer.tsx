/**
 * フッター
 * @module Footer
 */
import React from 'react'

/**
 * フッターのコンポーネントのプロパティ
 */
interface FooterProps {
  /**
   * 子要素
   */
  children?: React.ReactNode
}

/**
 * フッターのコンポーネント
 * @param {FooterProps} props -  フッターコンポーネントのプロパティ
 * @returns {React.FC} -  フッターコンポーネント
 */
export const Footer: React.FC<FooterProps> = ({ children, ...footerProps }) => {
  return (
    <footer id="footer" className="mt-auto" {...footerProps}>
      {children}
    </footer>
  )
}
