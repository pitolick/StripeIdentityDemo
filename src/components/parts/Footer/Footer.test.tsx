import React from 'react'
import { render } from '@testing-library/react'
import { Footer } from './Footer'

describe('Footerコンポーネント', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(
      <Footer>
        <div>フッターの子要素</div>
      </Footer>,
    )

    expect(container).toMatchSnapshot()
  })
})
