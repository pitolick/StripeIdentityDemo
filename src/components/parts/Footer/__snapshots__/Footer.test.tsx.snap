// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footerコンポーネント 正しくレンダリングされること 1`] = `
<div>
  <footer
    class="mt-auto"
    id="footer"
  >
    <div>
      フッターの子要素
    </div>
  </footer>
</div>
`;
